Sources/Sentry/SentryNSDataSwizzling.m

```diff
@@ -0,0 +1,104 @@
+#import "SentryNSDataSwizzling.h"
+#import "SentryNSDataTracker.h"
+#import "SentrySwizzle.h"
+#import <SentryLog.h>
+#import <objc/runtime.h>
+
+@implementation SentryNSDataSwizzling
+
++ (void)start
+{
+    [SentryNSDataTracker.sharedInstance enable];
+    [self swizzleNSData];
+}
+
++ (void)stop
+{
+    [SentryNSDataTracker.sharedInstance disable];
+}
+
+// SentrySwizzleInstanceMethod declaration shadows a local variable. The swizzling is working
+// fine and we accept this warning.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshadow"
++ (void)swizzleNSData
```

Comment:
The following methods of [NSData](https://developer.apple.com/documentation/foundation/nsdata?language=objc) are missing:

- `dataWithContentsOfFile:`
- `dataWithContentsOfFile:options:error:`
- `dataWithContentsOfURL:`
- `dataWithContentsOfURL:options:error:`
- `initWithContentsOfURL:`
- `writeToURL:atomically:`
- `writeToURL:options:error:`

Reply:
Those methods uses any of the swizzled ones.

I will created tests for all this tests in case apple changes implementation.